lib/providers/quoters/RoutingApiQuoter.ts

```diff
@@ -40,6 +47,11 @@ export class RoutingApiQuoter implements Quoter {
       }
       log.error(e, 'RoutingApiQuoterErr');
       metrics.putMetric(`RoutingApiQuoterErr`, 1);
+
+      // throw on all non-4xx errors
+      if (!(e instanceof AxiosError && e.response?.status?.toString().startsWith('4'))) {
```

Comment:
can we also add unit tests to expect the 500 error to be surfaced?

Reply:
Added unit tests for the changes 👍